Update a.json

luxelego opened the pull request:
<!--
Your pull request will be reviewed automatically and by a human.

The manual review may take several days or weeks,
depending on the reviewer's availability and workload.
If you haven't received a comment on your pull request
and it wasn't merged either,
it just hasn't been reviewed yet.

---

Please ensure the automated reviews pass.
Follow the instructions provided, if necessary.
You can speed up the process
by [running some tests locally](https://packagecontrol.io/docs/submitting_a_package#Step_7).

You can request a review from @packagecontrol-bot
to manually trigger an automated review
if you don't need to push a new commit.
Do **NOT** open a new pull request!

In general, make sure you:

 1. Used `"tags": true` and not `"branch": "master"`
    (versioning docs: <https://packagecontrol.io/docs/submitting_a_package#Step_4>)
 2. Added a README to your repository so that users (and reviewers) 
    can understand what your package provides.

You should proceed with a short description of what the package does
and, in case one or multiple similar package already exists,
why you believe it is different and needed
below this line. -->

braver commented:
You might want to add the "theme" label so people can find your package more easily. Also, consider excluding the png images using [.gitattributes](https://github.com/SublimeLinter/SublimeLinter/blob/master/.gitattributes).

luxelego commented:
> You might want to add the "theme" label so people can find your package more easily. Also, consider excluding the png images using [.gitattributes](https://github.com/SublimeLinter/SublimeLinter/blob/master/.gitattributes).

made it, thank you

braver commented:
Oh, it's a color scheme, not a theme. In that case, add a "color scheme" label (not just in the name), so it's listed here: https://packagecontrol.io/browse/labels/color%20scheme

luxelego commented:
> Oh, it's a color scheme, not a theme. In that case, add a "color scheme" label (not just in the name), so it's listed here: https://packagecontrol.io/browse/labels/color%20scheme

Yes, I create label "color scheme" in Issue.

jwortmann commented:
I believe it was meant to add a "labels" key in your pull request:
```diff
	{
			"name": "Alpen Color Scheme",
			"details": "https://github.com/luxelego/alpen_color_scheme",
+                       "labels": ["color scheme"],
			"releases": [
				{
					"sublime_text": "*",
					"branch": "master"
				}
			]
		},
```

***

I took a quick look into your color scheme and I saw a few lines that look like as if an automatic conversion messed up with line endings \n in scopes somehow, e.g. at https://github.com/luxelego/alpen_color_scheme/blob/6c6aa82da161c78c48fd8d818a13ca01c69bf38e/alpen.sublime-color-scheme#L2757. Although that's probably not breaking anything, but you still might have a look at those lines and fix them.

Also there are countless scopes like `col_XXXXXXXX`, are they related to a plugin? And just out of personal interest (because I have a color scheme too), would you mind telling me which plugins use the following scopes from your scheme (or maybe aren't they actually used by Sublime Text?): `block_cursor`, `markup.prompt`, `markup.strikethrough`, `markup.traceback`, `wordhighlight`, `highlight.color.ADD8E6`

luxelego commented:
> I believe it was meant to add a "labels" key in your pull request:
> 
> ```diff
> 	{
> 			"name": "Alpen Color Scheme",
> 			"details": "https://github.com/luxelego/alpen_color_scheme",
> +                       "labels": ["color scheme"],
> 			"releases": [
> 				{
> 					"sublime_text": "*",
> 					"branch": "master"
> 				}
> 			]
> 		},
> ```
> 
> I took a quick look into your color scheme and I saw a few lines that look like as if an automatic conversion messed up with line endings \n in scopes somehow, e.g. at https://github.com/luxelego/alpen_color_scheme/blob/6c6aa82da161c78c48fd8d818a13ca01c69bf38e/alpen.sublime-color-scheme#L2757. Although that's probably not breaking anything, but you still might have a look at those lines and fix them.
> 
> Also there are countless scopes like `col_XXXXXXXX`, are they related to a plugin? And just out of personal interest (because I have a color scheme too), would you mind telling me which plugins use the following scopes from your scheme (or maybe aren't they actually used by Sublime Text?): `block_cursor`, `markup.prompt`, `markup.strikethrough`, `markup.traceback`, `wordhighlight`, `highlight.color.ADD8E6`

Thanks you, I insert "labels" in pull request and today I will change line endings in my color scheme. Yes, I used - https://tmtheme-editor.herokuapp.com/ for generate scheme and then convert it in Sublime in ..sublime-color-scheme format. Now I will rebuild my color scheme in in manual mode.

jwortmann commented:
Yeah, I assumed that it was converted somehow to the .sublime-color-scheme format. This format actually makes it very easy to work with (for example you can use variables for colors) and there is a good documentation at https://www.sublimetext.com/docs/color_schemes.html. For the scope names used by language syntaxes there is an extensive list at https://www.sublimetext.com/docs/scope_naming.html and the `Tools > Developer > Show Scope Name` command can be very helpful.

luxelego commented:
> Yeah, I assumed that it was converted somehow to the .sublime-color-scheme format. This format actually makes it very easy to work with (for example you can use variables for colors) and there is a good documentation at https://www.sublimetext.com/docs/color_schemes.html. For the scope names used by language syntaxes there is an extensive list at https://www.sublimetext.com/docs/scope_naming.html and the `Tools > Developer > Show Scope Name` command can be very helpful.

Thanks a lot, yours Light Brackets Color Scheme - one is my favorite 👍

luxelego commented:
I remaded Alpen Color Scheme, there were 3000 lines of code, now 1000, Size was 100kb, now 34kb.

braver commented:
Looks good, but please fix the sorting issue. Also, you intend to merge 2 color schemes here right?

braver commented:
Oh good catch, I tend to forget to even look for that

luxelego commented:
> Looks good, but please fix the sorting issue. Also, you intend to merge 2 color schemes here right?

I don`t know how to merge 2 color schemes in one file.  Sorting fixed.

braver commented on repository/l.json:
```diff
@@ -1604,6 +1604,17 @@
+			"releases": [
+				{
+					"sublime_text": "*",
+					"branch": "master"
```
Please change this to "tags": true so that package control will pick up your tags, not simply every change to master (which is deprecated anyway)

braver commented:
>I don`t know how to merge 2 color schemes in one file. Sorting fixed.

Sorry, wasn't clear. Usually one PR handles one package, but yours here has two. Just making sure that's not by accident.

luxelego commented:
> > I don`t know how to merge 2 color schemes in one file. Sorting fixed.
> 
> Sorry, wasn't clear. Usually one PR handles one package, but yours here has two. Just making sure that's not by accident.
Everything is fine, I added the dark theme separately, but it appeared in this thread, I don't know why. I'm not a big pro at GitHub

luxelego commented:
> ### Automated testing result: ERROR
> Repo links:
> 
> * [Alpen Color Scheme](https://github.com/luxelego/alpen_color_scheme)
> * [Lotus Color Scheme](https://github.com/luxelego/lotus_color_scheme)
> 
> [Results help](https://github.com/packagecontrol/st_package_reviewer/wiki/Package-checks)
> 
> ```
> Packages added:
>   - Alpen Color Scheme
>   - Lotus Color Scheme
> 
> Processing package "Alpen Color Scheme"
>   - ERROR: Error downloading repository. HTTP error 404 downloading https://api.github.com/repos/luxelego/alpen_color_scheme/commits?sha=master&per_page=1.
> 
> Processing package "Lotus Color Scheme"
>   - ERROR: Error downloading repository. HTTP error 404 downloading https://api.github.com/repos/luxelego/lotus_color_scheme/commits?sha=master&per_page=1.
> ```

Fixed all

jwortmann commented:
> Everything is fine, I added the dark theme separately, but it appeared in this thread, I don't know why.

That is because you added all commits to a single branch (master) of your fork. A typical way to work with Git and GitHub would be to create separate branches for each self contained change first (i.e. one branch for your Alpen scheme and another branch for the Lotus scheme) then checkout a branch, make the corresponding commit on that branch and then create a separate pull request from each branch.

The tag `v.1.0.0` in your Lotus color scheme will not work, it must be `v1.0.0` or `1.0.0` or similar ([semantic version](https://semver.org/)).

Also you could consider to rename the files to start with a capital letter if you don't want them to be shown as "alpen" and "lotus" in the command palette.

rwols commented:
This section in the README is confusing: https://github.com/luxelego/alpen_color_scheme#activate-theme

The section is called "Activate Theme", but it should be called "Activate Color Scheme".

(same for https://github.com/luxelego/lotus_color_scheme#activate-theme)

luxelego commented:
> Activate Color Scheme

Thank you, fixed Activate Color Scheme in both repo.

luxelego commented:
> > Everything is fine, I added the dark theme separately, but it appeared in this thread, I don't know why.
> 
> That is because you added all commits to a single branch (master) of your fork. A typical way to work with Git and GitHub would be to create separate branches for each self contained change first (i.e. one branch for your Alpen scheme and another branch for the Lotus scheme) then checkout a branch, make the corresponding commit on that branch and then create a separate pull request from each branch.
> 
> The tag `v.1.0.0` in your Lotus color scheme will not work, it must be `v1.0.0` or `1.0.0` or similar ([semantic version](https://semver.org/)).
> 
> Also you could consider to rename the files to start with a capital letter if you don't want them to be shown as "alpen" and "lotus" in the command palette.

Fixed relize

braver commented:
> Also you could consider to rename the files to start with a capital letter if you don't want them to be shown as "alpen" and "lotus" in the command palette.

@jwortmann the color schemes have a "name" property, I kinda expect ST to use that and not the filename (although I haven't tested that assumption).

luxelego commented:
> > Also you could consider to rename the files to start with a capital letter if you don't want them to be shown as "alpen" and "lotus" in the command palette.
> 
> @jwortmann the color schemes have a "name" property, I kinda expect ST to use that and not the filename (although I haven't tested that assumption).
> 
> Edit: now I did test it and apparently it _does_ take the filename, not the name property. That's a bit lame, but there is never any harm in giving your files the proper name. @luxelego you might want to update your filenames.

You a right, I change in both repo - names with a capital letters.
